packages/sdk/typescript/human-protocol-sdk/test/escrow.test.ts

```
@@ -0,0 +1,1368 @@
+      const escrowAddress = FAKE_ADDRESS;
+
+      await expect(escrowClient.getTokenAddress(escrowAddress)).rejects.toThrow(
+        ErrorInvalidEscrowAddressProvided
+      );
+    });
+
+    test('should throw an error if an hasEscrow return false', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(false);
+
+      await expect(escrowClient.getTokenAddress(escrowAddress)).rejects.toThrow(
+        ErrorEscrowAddressIsNotProvidedByFactory
+      );
+    });
+
+    test('should successfully getTokenAddress escrow', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.escrowContract.token.mockReturnValue(
+        ethers.constants.AddressZero
+      );
+
+      const tokenAddress = await escrowClient.getTokenAddress(escrowAddress);
+
+      expect(tokenAddress).toEqual(ethers.constants.AddressZero);
+      expect(escrowClient.escrowContract.token).toHaveBeenCalledWith();
+    });
+
+    test('should throw an error if the getTokenAddress fails', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.escrowContract.token.mockRejectedValueOnce(new Error());
+
+      await expect(
+        escrowClient.getTokenAddress(escrowAddress)
+      ).rejects.toThrow();
+
+      expect(escrowClient.escrowContract.token).toHaveBeenCalledWith();
+    });
+  });
+
+  describe('getStatus', () => {
+    test('should throw an error if escrowAddress is an invalid address', async () => {
+      const escrowAddress = FAKE_ADDRESS;
+
+      await expect(escrowClient.getStatus(escrowAddress)).rejects.toThrow(
+        ErrorInvalidEscrowAddressProvided
+      );
+    });
+
+    test('should throw an error if an hasEscrow return false', async () => {
```

fix writing:
'should throw an error if hasEscrow returns false'